chalk-solve/src/clauses.rs

```diff
@@ -377,6 +384,57 @@ fn program_clauses_that_could_match<I: Interner>(
     Ok(clauses)
 }
 
+/// Adds the clause:
+///
+/// forall<T1, T2> Normalize(<T1 as Trait>::Item -> T2)
+///     :- Implemented(T1: Trait), Compatible, DownstreamType(T1), CannotProve
```

Comment:
I'm not sure if this is the right way to fix this coherence problem, but it seems to be working.